code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -0,0 +1,110 @@
+//I JUST WANNA GRILL FOR GOD'S SAKE
+
+/obj/machinery/grill
+	name = "grill"
+	desc = "Just like the old days."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "grill_open"
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/grill_fuel = 0
+	var/grilled_item
+	var/datum/looping_sound/grill/grill_loop
+
+/obj/machinery/grill/Initialize()
+	..()
+	grill_loop = new(list(src), FALSE)
+
+/obj/machinery/grill/update_icon()
+	if(grilled_item)
+		icon_state = "grill"
+	else if(!grill_fuel)
+		icon_state = "grill_open"
+	else
+		icon_state = "grill_on"
+
+/obj/machinery/grill/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/trash/coal))
+		qdel(I)
+		to_chat(user, "<span class='notice'>You put the [I] in [src].</span>")
+		add_fuel(500)
+		return
+	if(istype(I, /obj/item/stack/sheet/mineral/wood))
+		qdel(I)
+		to_chat(user, "<span class='notice'>You put the [S.amount] [I]s in [src].</span>")
+		var/obj/item/stack/S = I
+		add_fuel(50 * S.amount)
```

Comment:
Use mat datums

Reply:
how

Reply:
wood and coal arent even datum mats and im not bothering enough to make them